servers/services/src/test/java/org/projectnessie/services/impl/RefUtilTest.java

```diff
@@ -0,0 +1,137 @@
+import org.projectnessie.versioned.NamedRef;
+import org.projectnessie.versioned.TagName;
+
+class RefUtilTest {
+
+  public static final String HASH_VALUE = "deadbeeffeedcafe";
+  public static final String REF_NAME = "foo";
+
+  @Test
+  void toNamedRef() {
+    assertThat(RefUtil.toNamedRef(Branch.of(REF_NAME, HASH_VALUE)))
+        .isEqualTo(BranchName.of(REF_NAME));
+    assertThat(RefUtil.toNamedRef(Tag.of(REF_NAME, HASH_VALUE))).isEqualTo(TagName.of(REF_NAME));
+  }
+
+  @Test
+  void toNamedRefErrors() {
+    assertThatThrownBy(() -> RefUtil.toNamedRef(null)).isInstanceOf(NullPointerException.class);
+    assertThatThrownBy(
+            () ->
+                RefUtil.toNamedRef(
+                    new Reference() {
+                      @Override
+                      public String getName() {
+                        return null;
+                      }
+
+                      @Override
+                      public String getHash() {
+                        return null;
+                      }
+
+                      @Override
+                      public ReferenceMetadata getMetadata() {
+                        return null;
+                      }
+                    }))
+        .isInstanceOf(IllegalArgumentException.class);
+  }
+
+  @Test
+  void toReferenceString() {
+    assertThat(RefUtil.toReference(BranchName.of(REF_NAME), HASH_VALUE))
+        .isEqualTo(Branch.of(REF_NAME, HASH_VALUE));
+    assertThat(RefUtil.toReference(TagName.of(REF_NAME), HASH_VALUE))
+        .isEqualTo(Tag.of(REF_NAME, HASH_VALUE));
+  }
+
+  @Test
+  void toReferenceHash() {
+    assertThat(RefUtil.toReference(BranchName.of(REF_NAME), Hash.of(HASH_VALUE)))
+        .isEqualTo(Branch.of(REF_NAME, HASH_VALUE));
+    assertThat(RefUtil.toReference(TagName.of(REF_NAME), Hash.of(HASH_VALUE)))
+        .isEqualTo(Tag.of(REF_NAME, HASH_VALUE));
+  }
+
+  @Test
+  void toReferenceStringErrors() {
+    assertThatThrownBy(() -> toReference(null, (String) null))
+        .isInstanceOf(NullPointerException.class);
+    assertThatThrownBy(() -> toReference(BranchName.of(REF_NAME), (String) null))
+        .isInstanceOf(NullPointerException.class);
+    assertThatThrownBy(() -> toReference(TagName.of(REF_NAME), (String) null))
+        .isInstanceOf(NullPointerException.class);
+    assertThatThrownBy(() -> toReference(null, HASH_VALUE))
+        .isInstanceOf(NullPointerException.class);
+    assertThatThrownBy(
+            () ->
+                toReference(
+                    new NamedRef() {
+                      @Override
+                      @Nonnull
+                      public String getName() {
+                        return REF_NAME;
+                      }
+                    },
+                    HASH_VALUE))
+        .isInstanceOf(IllegalArgumentException.class);
+    assertThatThrownBy(() -> toReference(() -> REF_NAME, HASH_VALUE))
+        .isInstanceOf(IllegalArgumentException.class);
```

Comment:
nit: would be good to check that we get the correct error messages. Same for all the other places where things are verified via `assertThatThrownBy`